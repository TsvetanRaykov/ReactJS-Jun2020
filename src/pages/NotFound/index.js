import React from 'react'
import { Paper, Avatar, Typography, makeStyles } from '@material-ui/core'

const useStyles = makeStyles((theme) => ({
	main: {
		width: 'auto',
		display: 'block',
		marginLeft: theme.spacing(3),
		marginRight: theme.spacing(3),
		[theme.breakpoints.up(400 + theme.spacing(3 * 2))]: {
			width: 400,
			marginLeft: 'auto',
			marginRight: 'auto',
		},
	},
	paper: {
		marginTop: theme.spacing(8),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
		padding: `${theme.spacing(2)}px ${theme.spacing(3)}px ${theme.spacing(
			3
		)}px`,
	},
	avatar: {
		margin: theme.spacing(2),
		backgroundColor: theme.palette.secondary.main,
		width: theme.spacing(7),
		height: theme.spacing(7),
	},
}))

const NotFound = () => {
	const classes = useStyles()
	return (
		<main className={classes.main}>
			<Paper className={classes.paper}>
				<Avatar className={classes.avatar} src='/logo.jpg' />
				<br />
				<Typography component='h1' variant='h5'>
					Content Not Found
				</Typography>
			</Paper>
		</main>
	)
}

export default NotFound
