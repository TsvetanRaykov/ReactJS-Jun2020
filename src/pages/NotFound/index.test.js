import React from 'react'
import renderer from 'react-test-renderer'
import TestWrapper from '../../utils/test-wrapper'
import NotFound from '.'

jest.mock('react', () => ({
	...jest.requireActual('react'),
	useLayoutEffect: jest.requireActual('react').useEffect,
}))

describe('<NotFound />', () => {
	it('Should match the snapshot', () => {
		const wrapper = renderer.create(
			<TestWrapper>
				<NotFound />
			</TestWrapper>
		)
		expect(wrapper).toMatchSnapshot()
	})
})
