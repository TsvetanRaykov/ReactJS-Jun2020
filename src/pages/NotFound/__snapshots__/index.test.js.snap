// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Should match the snapshot 1`] = `
<main
  className="makeStyles-main-1"
>
  <div
    className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-3"
    >
      <img
        className="MuiAvatar-img"
        src="/logo.jpg"
      />
    </div>
    <br />
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Content Not Found
    </h1>
  </div>
</main>
`;
